pkg/network/ovn_kubernetes.go

```diff
@@ -561,12 +579,12 @@ func shouldRenderIPsec(conf *operv1.OVNKubernetesConfig, bootstrapResult *bootst
 	// MachineConfig IPsec extensions are needed for the ipsec deployment except
 	// when the containerized deployment is used in hypershift hosted clusters.
 	// We will rollout unless the user has rolled out its own.
-	renderIPsecMachineConfig = renderIPsecDaemonSet && !isUserIPsecMachineConfigPresent && !isHypershiftHostedCluster
+	renderIPsecMachineConfig = (isIPsecEnabled || renderIPsecDaemonSet) && !isUserIPsecMachineConfigPresent && !isHypershiftHostedCluster
 
 	// We render OVN IPsec if IPsec is enabled or it's upgrade is in progress.
```

Comment:
I believe the comment here is misleading and we should change it to 

// We render OVN IPsec if **EW** IPsec is enabled or it's upgrade is in progress.

Alternatively the code should be changed - (in line 587 it should be **isIPsecFull**